### PR TITLE
Track registration source on-chain via ENSIP 14

### PR DESCRIPTION
**Pull request by gskril**
When generating a secret for registering names, we can use the beginning of the enslabs.eth namehash to track on-chain what portion of registrations are coming from the official manager app as outlined in [ENSIP 14](https://github.com/ensdomains/docs/pull/127).

Before:
```js
'0x' + crypto.randomBytes(32).toString('hex')
```

After:
```js
const platformSource = '9923eb94'
'0x' + platformSource + crypto.randomBytes(28).toString('hex')
```

### CODE REVIEW

**Comment by makoto** [comment]
```
const platformSource = '9923eb94'
```

How is this derived? The draft ENSIP doesn't mention how the code is assigned or managed.

I suggest not merging this until the ENSIP itself get merged and the official number is assigned (there is actually another 14 in the PR) https://github.com/ensdomains/docs/pull/127/files.

**Review by makoto — CHANGES_REQUESTED** [review]
Isn't is more explicit to put the name of the site like `app.ens.domains`? Question remains what to do with IPFS version though

**Comment by gskril** [comment]
> How is this derived? The draft ENSIP doesn't mention how the code is assigned or managed.

The proposal includes a line that "suggests a social convention that generates the secret from a combination of taking the first 6 characters of a namehash...". So `9923eb94` is the first 8 characters (just to be safe since that was a suggested change to the proposal) of enslabs.eth.

But generally I agree this can wait until the proposal itself is merged :)

**Pull request closed without merging**
